### PR TITLE
Adjust roadmap items and outcomes

### PR DESCRIPTION
**Pull request by macraig**
Outcomes are now reflected per milestone except for the two more high-level outcomes that would be reached by completing our overall goal.
Based on the team's research on auto-configuring cmake repos on Cloud, we've rephrased and added steps to our roadmap items to better reflect our efforts and expected outcomes.

### CODE REVIEW

File: handbook/engineering/code-intelligence/goals.md [hunk 1]
```diff
@@ -12,48 +12,57 @@ Progress on adoption and usage is tracked in our [Looker dashboard](https://sour
...
-    - Auto-index 5 repositories on Sourcegraph Cloud that use Bazel as a build system
-    - Add inference rules for Bazel projects and have at least 5 repositories fully auto-indexed on Cloud
+    - ✅ Increase observability in the code intel executor to make troubleshooting failed auto-index records possible.
+    - ✅ Auto-index 5 repositories on Sourcegraph Cloud.
+    
+    **Outcome:** 5 popular open source C++ repositories have up-to-date precise code intelligence on Sourcegraph Cloud.
+
+1. 🔄 Sourcegraph Cloud has precise code intelligence for C++ repositories that use cmake as a build system. 
+    - 🔄 Auto-index 5 repositories on Sourcegraph Cloud that use cmake as a build system.
+    - ✅ Research: Add inference rules for cmake projects to auto-index on Cloud.
```
Comment (efritz) [inline handbook/engineering/code-intelligence/goals.md:30]: :cool:

File: handbook/engineering/code-intelligence/goals.md [hunk 2]
```diff
@@ -12,48 +12,57 @@ Progress on adoption and usage is tracked in our [Looker dashboard](https://sour
...
+    - 🔄 Auto-detect C++/cmake repositories and display common configurations and documentation to guide users in their set-up.
```
Comment (efritz) [inline handbook/engineering/code-intelligence/goals.md:31]: Yeah I think this is a better goal (and outcome) than having succeeded in the research.

**Review by efritz — APPROVED** [review]

File: handbook/engineering/code-intelligence/goals.md [hunk 3]
```diff
@@ -12,48 +12,57 @@ Progress on adoption and usage is tracked in our [Looker dashboard](https://sour
...
 1. Sourcegraph Cloud has up-to-date precise code intelligence for C++ repositories that exceed 1mil SLOC.
-    - Reduce lsif-clang memory usage
-    - Auto-index 5 repositories on Sourcegraph Cloud that exceed 1mil SLOC
+    - Reduce lsif-clang memory usage.
+    - Auto-index 5 repositories on Sourcegraph Cloud that exceed 1M SLOC.
+    
+    **Outcome:** 5 popular open source C++ repositories that exceed 1M SLOC have up-to-date precise code intelligence on Sourcegraph Cloud.
 
-1. Increase precise code intel operations by ([*N<sub>1</sub>*][N1]+[*N<sub>2</sub>*][N2]+[*N<sub>3</sub>*][N3]) per month.
+1. Increase precise code intel montly operations.
```
Comment (scalabilitysolved) [inline handbook/engineering/code-intelligence/goals.md:49]:
```suggestion
1. Increase precise code intel monthly operations.
```

**Review by scalabilitysolved — APPROVED** [review]
Looks good, one auto-suggest correction :balloon:

**Review by garobrik — APPROVED** [review]

File: handbook/engineering/code-intelligence/goals.md (hunk 3 above)
Comment (macraig) [reply handbook/engineering/code-intelligence/goals.md]: Thanks for catching!

**Pull request merged**